tests/unit/Entity/EntityIdValueTest.php

```diff
@@ -29,26 +32,25 @@ public function testCanConstruct() {
 	 * @dataProvider instanceProvider
 	 */
 	public function testSerialzationRoundtrip( EntityIdValue $id ) {
-		$newId = unserialize( serialize( $id ) );
+		$serialized = serialize( $id );
+		$newId = unserialize( $serialized );
 
 		$this->assertEquals( $id, $newId );
 	}
 
 	public function instanceProvider() {
 		$ids = [
-			new ItemId( 'Q1' ),
-			new ItemId( 'Q42' ),
-			new ItemId( 'Q31337' ),
-			new ItemId( 'Q2147483647' ),
-			new PropertyId( 'P1' ),
-			new PropertyId( 'P42' ),
-			new PropertyId( 'P31337' ),
+			'Q1' => new ItemId( 'Q1' ),
+			'Q2147483647' => new ItemId( 'Q2147483647' ),
+			'P31337' => new PropertyId( 'P31337' ),
+			'X567' => $this->newCustomId( 'X567' ),
+			'foo:P678' => new PropertyId( 'foo:P678' ),
```

Comment:
Sorry. What I meant is that the *removal* of test cases is unrelated and should be done in a separate patch.